pkg/packagekit/updates.jsx

```diff
@@ -785,6 +818,9 @@ class OsUpdates extends React.Component {
                 </div>);
         }
 
+        if (["available", "uptodate", "updateSuccess"].indexOf(this.state.state) == -1)
+            page_status.set_own(null);
```

Comment:
This looks backwards -- aren't these precisely the "good" states where we *do* want to clear the notification? I. e. shouldn't this be `>= 0`? Or did I misunderstand what `set_own(null)` does? In the latter case this should have a comment.

Reply:
These are the states that call `page_status.set_own()` themselves further down.  An alternative would be to insert `set_own(null)` into all cases of the switch.  That might indeed be less confusing.

Reply:
I have done that, and I like it much better.

Reply:
That avoids the indexOf(), but I'm still wondering the same thing: Why do the "bad" states clear the notification, and the "good" ones don't? It feels like it should be the other way around.

Reply:
A page status isn't always bad.  For example, "uptodate" sets the status to "System is up to date" (without any icon in nav menu).

The "loadError" state could set a state if we want it to.  Same for the other states, so let's go down the list.

"loading", "refreshing", "locked": I think the page status can be silent about these intermediate states.

"loadError", "updateError": These could set the page status so that the user is guided back to the Updates page to review the error messages.

"applying": Should be silent, I'd say.

"restart": Silent as well.

What do you think?

Reply:
Right, makes sense. It's much less confusing now in the current PR, thanks!